Snakefile

```diff
@@ -153,21 +155,113 @@ rule iq_tree:
         rooted="output/strains/{strain}/besttree.nwk",
         unrooted=temp("output/strains/{strain}/besttree.unrooted.treefile"),
         iqtree=temp("output/strains/{strain}/besttree.unrooted.iqtree"),
-        ckp=temp("output/strains/{strain}/besttree.unrooted.ckp.gz")
+        ckp=temp("output/strains/{strain}/besttree.unrooted.ckp.gz"),
     log:
         "output/strains/{strain}/besttree.unrooted.log"
     params:
         enabled=config['iqtree']['enabled'],
         mode=config['iqtree']['mode'],
         model=config['iqtree']['model'],
-        prefix="output/strains/{strain}/besttree.unrooted"
+        prefix="output/strains/{strain}/besttree.unrooted",
+        temp="output/strains/{strain}/temp_align_variants.aln"
+
     conda:
         config["poppipe_location"] + "/envs/iqtree.yml"
     threads:
-        16
+        4
     script:
         config["poppipe_location"] + "/scripts/run_iqtree.py"
 
+rule gubbins:
+    input:
+        alignment="output/strains/{strain}/align_variants.aln",
+        start_tree="output/strains/{strain}/besttree.nwk"
+    output:
+        final_tree="output/strains/{strain}/gubbins.final_tree.tre",
+    group:
+        "gubbins"
+    # log:
+    #     "logs/gubbins_{strain}.log"
+    params:
+        prefix=config['gubbins']['prefix'],
+        tree_builder=config['gubbins']['tree-builder'],
+        min_snp=config['gubbins']['min-snps'],
+        min_window=config['gubbins']['min-window-size'],
+        max_window=config['gubbins']['max-window-size'],
+        iterations=config['gubbins']['iterations'],
+        directory=config['gubbins']['dir'],
+        aln="align_variants.aln",
+        btree="besttree.nwk"
+
+    conda:
+        config["poppipe_location"] + "/envs/gubbins.yml"
+    threads:
+        4
+    shell:
+        "pushd {params.directory} &&\
+        run_gubbins.py {params.aln} --prefix {params.prefix} --starting-tree {params.btree} --threads {threads} \
+        && popd"
+
+rule bactdating:
+    input:
+        final_tree="output/strains/{strain}/gubbins.final_tree.tre",
```

Comment:
You give the tree as input, but then use the whole gubbins directory in the command below? Would be better to use `{input.final_tree}` rather than generating this path twice

Reply:
I had it that way because gubbins is already looking for `final_tree.tre` so when I give `{input.final_tree}` as an input it does not work because: `cannot open file 'output/strains/10/gubbins.final_tree.tre.final_tree.tre': No such file or directory`

Reply:
I mean by changing your `scripts/run_bactDating.R` to use this input rather than the full path, if that works

Reply:
I am using the `loadGubbins` function in `scripts/run_bactDating.R` which automatically assumes that the name of the file is `final_tree.tre`but I'll check in the gubbins doc if it is possible to change that

Reply:
Ah I see, probably not then! In that case just add a comment into the snakefile and/or Rscript explaining this

Reply:
Seems like the [code](https://github.com/xavierdidelot/BactDating/blob/0e93c298ffe3b3006c06d95f40091889a2f6d1e7/R/loadGubbins.R#L7) is wanting only the prefix so then that wouldn't work